packages/git/src/browser/git-contribution.ts

```diff
@@ -440,6 +441,19 @@ export class GitContribution implements CommandContribution, MenuContribution, T
         registry.registerCommand(GIT_COMMANDS.OPEN_CHANGED_FILE, {
             execute: (...arg: ScmResource[]) => {
                 for (const resource of arg) {
+                    const provider = this.repositoryProvider.selectedScmProvider;
+                    if (provider) {
+                        const gitStatus = provider.getStatus();
+                        if (gitStatus) {
+                            const deletedFile = gitStatus.changes.find(change =>
+                                change.status === GitFileStatus.Deleted && change.uri === resource.sourceUri.toString());
+                            if (deletedFile) {
+                                const deletedFileUri = new URI(deletedFile.uri).withScheme(GIT_RESOURCE_SCHEME).withQuery('HEAD');
+                                this.editorManager.open(deletedFileUri, { mode: 'reveal' });
+                                return;
+                            }
+                        }
+                    }
                     this.editorManager.open(resource.sourceUri, { mode: 'reveal' });
```

Comment:
could you try to call `resource.open` here instead and revert the rest

Reply:
`resource.open()` will open the diff editor in the `Modified` case

Reply:
The key is how to filter the `Deleted` cases by `resource`. I know my current method is not that elegant. Any suggestions?

Reply:
let's go with it if it works